src/agents/utils/signatures.ts

```diff
@@ -0,0 +1,139 @@
+  types: Record<string, TypedDataField[]>,
+  message: Record<string, any>,
+  signature: string
+) => Promise<boolean>;
+
+type VerifyOptions = {
+  /** Allow ECDSA signatures. */
+  ecdsa?: boolean;
+  /** Allow EIP-1271 signatures. */
+  eip1271?: boolean;
+};
+
+const EVM_TYPED_DATA_CHAIN_ID = parseInt(
+  process.env.TYPED_DATA_CHAIN_ID ?? '11155111'
+);
+
+const ERC1271_ABI = [
+  'function isValidSignature(bytes32 _hash, bytes memory _signature) public view returns (bytes4 magicValue)'
+];
+const ERC1271_ABI_OLD = [
+  'function isValidSignature(bytes _hash, bytes memory _signature) public view returns (bytes4 magicValue)'
+];
+const ERC1271_MAGIC_VALUE = '0x1626ba7e';
+const ERC1271_MAGIC_VALUE_OLD = '0x20c13b0b';
+
+const DOMAIN: TypedDataDomain = {
+  name: 'highlight',
+  version: '0.1.0'
+};
+
+const provider = new StaticJsonRpcProvider(
+  `https://rpc.snapshot.org/${EVM_TYPED_DATA_CHAIN_ID}`,
+  EVM_TYPED_DATA_CHAIN_ID
+);
+
+function isEqual(a: string, b: string): boolean {
+  return a.toLowerCase() === b.toLowerCase();
+}
+
+function getHash(data: Record<string, any>): string {
+  const { types, message } = data;
+  return _TypedDataEncoder.hash(DOMAIN, types, message);
+}
+
+export async function signMessage(
+  wallet: Wallet,
+  types: Record<string, TypedDataField[]>,
+  message: Record<string, any>
+): Promise<string> {
+  return wallet._signTypedData(DOMAIN, types, message);
+}
+
+export const verifyEcdsaSignature: SignatureVerifier = async (
+  address,
+  types,
+  message,
+  signature
+) => {
+  const recoveredAddress = verifyTypedData(DOMAIN, types, message, signature);
+
+  return isEqual(recoveredAddress, address);
+};
+
+export const verifyEip1271Signature: SignatureVerifier = async (
+  address,
+  types,
+  message,
+  signature
+) => {
+  const hash = getHash({ types, message });
+
+  const params = [address, signature, hash] as const;
+
+  const valid = await verifyEip1271SignatureWithAbi(
+    ERC1271_ABI,
+    ERC1271_MAGIC_VALUE,
+    ...params
+  );
+  if (valid) return true;
+
+  const validOld = await verifyEip1271SignatureWithAbi(
+    ERC1271_ABI_OLD,
+    ERC1271_MAGIC_VALUE_OLD,
+    ...params
+  );
+  if (validOld) return true;
+  return false;
+};
+
+async function verifyEip1271SignatureWithAbi(
+  abi: ContractInterface,
+  magicValue: string,
+  address: string,
+  sig: string,
+  hash: string
+): Promise<boolean> {
+  let returnValue: string;
+  try {
+    const contract = new Contract(address, abi, provider);
+    returnValue = await contract.isValidSignature(hash, sig);
+  } catch {
+    return false;
+  }
+
+  return isEqual(returnValue, magicValue);
+}
+
+export async function verifySignature(
+  address: string,
+  types: Record<string, TypedDataField[]>,
+  message: Record<string, any>,
+  signature: string,
+  options: VerifyOptions
+) {
+  const params = [address, types, message, signature] as const;
+
+  if (options.ecdsa) {
+    const valid = await verifyEcdsaSignature(...params);
```

Comment:
We don't need a promise here
```suggestion
    const valid = verifyEcdsaSignature(...params);
```

Reply:
Not sure if it was like this before, but all those functions are using `SignatureVerifier` type that returns `Promise<boolean>` and are async.